### PR TITLE
openexr & ilmbase 2.4.0

### PR DESCRIPTION
**Pull request by equal-l2**
Created with `brew bump-formula-pr`.

### CODE REVIEW

File: Formula/openexr.rb [hunk 1]
```diff
@@ -1,8 +1,8 @@
 class Openexr < Formula
   desc "High dynamic-range image file format"
   homepage "https://www.openexr.com/"
-  url "https://github.com/openexr/openexr/releases/download/v2.3.0/openexr-2.3.0.tar.gz"
-  sha256 "fd6cb3a87f8c1a233be17b94c74799e6241d50fc5efd4df75c7a4b9cf4e25ea6"
+  url "https://github.com/openexr/openexr/archive/v2.4.0.tar.gz"
```
Comment (fxcoudert) [inline Formula/openexr.rb]: There is no `configure` in that tarball

Comment (equal-l2) [reply Formula/openexr.rb]: Fixed by using cmake, thanks!

**Comment by equal-l2** [comment]
It seems that `ilmbase` now requires C++11.
@fxcoudert How can I handle this?

**Comment by equal-l2** [comment]
Adding `ENV.cxx11` to all dependents?

**Comment by SMillerDev** [comment]
@equal-l2 could you rebase on master? And isn't C11 the default nowadays?

**Comment by equal-l2** [comment]
pkgconfig file generated by `openexr` seems broken.
https://github.com/openexr/openexr/issues/567

**Comment by equal-l2** [comment]
`ctl` has something wrong in its linking phase:
```
[ 46%] Linking CXX shared library libIlmCtlMath.dylib
cd /tmp/ctl-20191112-70143-1hy43yz/CTL-ctl-1.5.2/build/lib/IlmCtlMath && /usr/local/Cellar/cmake/3.15.5/bin/cmake -E cmake_link_script CMakeFiles/IlmCtlMath.dir/link.txt --verbose=1
/usr/local/Homebrew/Library/Homebrew/shims/mac/super/clang++   -DNDEBUG -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk -dynamiclib -Wl,-headerpad_max_install_names   -o libIlmImfCtl.dylib -install_name /tmp/ctl-20191112-70143-1hy43yz/CTL-ctl-1.5.2/build/lib/IlmImfCtl/libIlmImfCtl.dylib CMakeFiles/IlmImfCtl.dir/ImfCtlApplyTransforms.cpp.o CMakeFiles/IlmImfCtl.dir/ImfCtlCopyFunctionArg.cpp.o  -L/usr/local/Cellar/ilmbase/2.4.0/lib  -L/usr/local/Cellar/openexr/2.4.0/lib -Wl,-rpath,/usr/local/Cellar/ilmbase/2.4.0/lib -Wl,-rpath,/usr/local/Cellar/openexr/2.4.0/lib ../IlmCtl/libIlmCtl.1.5.0.dylib -lIlmImf -lIex -lIlmThread -lHalf 
/usr/local/Homebrew/Library/Homebrew/shims/mac/super/clang++   -DNDEBUG -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk -dynamiclib -Wl,-headerpad_max_install_names   -compatibility_version 1.5.0 -current_version 1.5.0 -o libIlmCtlMath.1.5.0.dylib -install_name /tmp/ctl-20191112-70143-1hy43yz/CTL-ctl-1.5.2/build/lib/IlmCtlMath/libIlmCtlMath.1.5.0.dylib CMakeFiles/IlmCtlMath.dir/CtlColorSpace.cpp.o CMakeFiles/IlmCtlMath.dir/CtlLookupTable.cpp.o CMakeFiles/IlmCtlMath.dir/CtlRbfInterpolator.cpp.o  -L/usr/local/Cellar/ilmbase/2.4.0/lib -Wl,-rpath,/usr/local/Cellar/ilmbase/2.4.0/lib ../IlmCtl/libIlmCtl.1.5.0.dylib -lIex -lIlmThread -lHalf 
Undefined symbols for architecture x86_64:
  "Imath_2_4::SingMatrixExc::SingMatrixExc(char const*)", referenced from:
      Imath_2_4::Matrix44<float>::inverse(bool) const in CtlColorSpace.cpp.o
      Imath_2_4::Matrix44<float>::gjInverse(bool) const in CtlColorSpace.cpp.o
  "Imath_2_4::SingMatrixExc::~SingMatrixExc()", referenced from:
      Imath_2_4::Matrix44<float>::inverse(bool) const in CtlColorSpace.cpp.o
      Imath_2_4::Matrix44<float>::gjInverse(bool) const in CtlColorSpace.cpp.o
  "typeinfo for Imath_2_4::SingMatrixExc", referenced from:
      Imath_2_4::Matrix44<float>::inverse(bool) const in CtlColorSpace.cpp.o
      Imath_2_4::Matrix44<float>::gjInverse(bool) const in CtlColorSpace.cpp.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [lib/IlmCtlMath/libIlmCtlMath.1.5.0.dylib] Error 1
make[1]: *** [lib/IlmCtlMath/CMakeFiles/IlmCtlMath.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 46%] Built target IlmImfCtl
make: *** [all] Error 2
```

`field3d` (SCons) seems to be ignoring `ENV.cxx11`.
```
Compiling [1mbuild/darwin/m64/release/FieldCache.os[0m
In file included from src/ClassFactory.cpp:44:
In file included from export/ClassFactory.h:50:
In file included from export/Field.h:55:
In file included from export/Traits.h:54:
In file included from export/Types.h:56:
In file included from export/StdMathLib.h:43:
In file included from /usr/local/include/OpenEXR/ImathColor.h:48:
/usr/local/include/OpenEXR/half.h:99:15: warning: defaulted function definitions are a C++11 extension [-Wc++11-extensions]
    half () = default;                  // no initialization
              ^
/usr/local/include/OpenEXR/half.h:102:16: warning: defaulted function definitions are a C++11 extension [-Wc++11-extensions]
    ~half () = default;
               ^
/usr/local/include/OpenEXR/half.h:103:27: warning: defaulted function definitions are a C++11 extension [-Wc++11-extensions]
    half (const half &) = default;
                          ^
/usr/local/include/OpenEXR/half.h:104:16: warning: rvalue references are a C++11 extension [-Wc++11-extensions]
    half (half &&) noexcept = default;
               ^
/usr/local/include/OpenEXR/half.h:104:19: error: expected ';' at end of declaration list
    half (half &&) noexcept = default;
                  ^
                  ;
/usr/local/include/OpenEXR/half.h:124:43: warning: defaulted function definitions are a C++11 extension [-Wc++11-extensions]
    half &              operator = (const half  &h) = default;
                                                      ^
/usr/local/include/OpenEXR/half.h:125:31: warning: rvalue references are a C++11 extension [-Wc++11-extensions]
    half &              operator = (half  &&h) noexcept = default;
                                          ^
/usr/local/include/OpenEXR/half.h:125:35: error: expected ';' at end of declaration list
    half &              operator = (half  &&h) noexcept = default;
                                              ^
                                              ;
```

**Comment by Bo98** [comment]
Just going to mark this as "do not merge" until we've merged in Python 3.8 (#45337), due to merge conflicts. Feel free to keep working on this though and we'll be able to help out where need be.

**Comment by fxcoudert** [comment]
CI does not handle merge commits well, please rebase on top of master.

**Comment by fxcoudert** [comment]
Also need a revision: `openimageio vips`

**Comment by equal-l2** [comment]
@fxcoudert
Thanks for pointing out, I'll revision them soon.

As for `ctl` and `field3d`, I know what is the problem but I have no idea how to fix them. Any help is welcome.

All I have to do is:
- `ctl` : The linking error is due to the lack of `libImath`, thus add `-lImath` to the linking flags
- `field3d` : All the error is due to using old C++ standard, thus tell scons to use C++11

**Comment by equal-l2** [comment]
Can I propose deletion of `ctl` and `field3d`, since both are unmaintained (judging from having no commit for over two years) and are blocking this PR?

**Comment by zbeekman** [comment]
> Can I propose deletion of `ctl` and `field3d`, since both are unmaintained (judging from having no commit for over two years) and are blocking this PR?

@equal-l2 can we apply https://github.com/imageworks/Field3D/pull/97 as part of this PR while we wait for upstream to merge? Also, if https://github.com/ampas/CTL/pull/73 resolves the `ctl` issue could that be applied as part of this PR as well?

**Comment by equal-l2** [comment]
@fxcoudert I fixed every problems mentioned.

**Comment by fxcoudert** [comment]
Excellent work! Thanks @equal-l2 for your contribution to Homebrew

**Pull request closed without merging**